src/test/java/com/squareup/javapoet/TypeSpecTest.java

```diff
@@ -413,9 +415,9 @@
         .addTypeVariable(t)
         .addTypeVariable(p)
         .addSuperinterface(Types.parameterizedType(Comparable.class, p))
-        .addField(FieldSpec.of(t, "label"))
-        .addField(FieldSpec.of(p, "x"))
-        .addField(FieldSpec.of(p, "y"))
```

Comment:
Yeah, I think we'll want to study the calling code to decide where overloads help.